### ISSUE TITLE
Rule Change: Refactor no-irregular-whitespace to be language-agnostic

### ISSUE THREAD

**Issue by techmannih** [eslint/eslint#20605]
### What rule do you want to change?

no-irregular-whitespace

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
const message = "Hello\u200BWorld"; // Contains zero-width space
/* This is a comment with a \u00A0 non-breaking space */
```

### What does the rule currently do for this code?

The rule currently identifies irregular whitespace by inspecting specific AST node types (such as Literal and TemplateElement) and comment nodes. It relies on the parser to identify these structures beforehand.

This tight coupling with the AST structure makes it harder to apply the same detection logic consistently across different file types or parsers that ESLint supports.

### What will the rule do after it's changed?

The rule could be refactored to detect irregular whitespace using token-based scanning or raw source analysis via sourceCode.getText().

By scanning the raw source text instead of relying on specific AST node types, the rule becomes more parser-agnostic and aligns with ESLint’s long-term goal of supporting language-agnostic rules.

This would allow the rule to work consistently across different syntaxes and file formats processed by ESLint (such as Markdown code blocks, JSON, or custom parsers) while maintaining the same detection logic.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by fasttime** [comment]
We already have an open issue for this at #19805. Let's continue the discussion there. Thanks!
